flang/lib/Lower/IO.cpp

```diff
@@ -428,16 +428,27 @@ static mlir::FuncOp getOutputFunc(mlir::Location loc,
     else if (width == 64)
       return getIORuntimeFunc<mkIOKey(OutputReal64)>(loc, builder);
   }
+  auto kindMap = fir::getKindMapping(builder.getModule());
   if (auto ty = type.dyn_cast<fir::ComplexType>()) {
```

Comment:
What is the motivation for changing complex type code?  To uniformly use bit widths, rather than byte widths, even though that's slightly more code?  If that code is changed, it would be nice to be consistent with the REAL and INTEGER sections of the code.  Either place all width definitions inside ifs or all outside; and consistently use either `width` or `bitWidth`.  (My preference is the former, shorter code in both cases.)

Reply:
The change is to use (bit) widths throughout. Formerly it was hardcoded to use front-end KINDs, which is not correct.

I'll switch the variable names to `width` here to be consistent. (Someone doing upstreaming changed `width` to `bitWidth` in another file, which I agree wasn't really necessary.)